app/challenge.go

```diff
@@ -0,0 +1,28 @@
+package app
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"app/internal/pow"
+	"app/internal/util"
+)
+
+// ChallengeHandler is a handler for the /challenge endpoint.
+func ChallengeHandler(w http.ResponseWriter, r *http.Request) {
+	ctx, err := util.NewContext(w, r)
+	if err != nil {
+		return
+	}
+
+	if err = util.ValidateRequestMethod(&ctx, "GET", ""); err != nil {
```

Comment:
See my comment on the other PR-- I was initially confused by the lack of `ctx.WriteStatusError` in this `if` block.